src/mapping/attractor_mapping_recurrences.jl

```diff
@@ -159,7 +181,21 @@ function basins_of_attraction(mapper::AttractorsViaRecurrences; show_progress =
             `basins_of_attraction(mapper)`."""
         ))
     end
-    grid = mapper.grid
+    
+    if (mapper.bsn_nfo.grid_nfo isa SubdivisionBasedGrid)
```

Comment:
I mean this lines in particular, in some way I need to pass maximal lvl grid here in case of SubdivisionBasedGrid 
but I am failing to construct maximal lvl grid automatically for multiple dimensions as a tuple, as also further `generate_ic_on_grid` expects tuple.

Reply:
The maximal grid level is given as a function to the `subdivision_level_matrix`. I still don't udnerstand your question. In this line in particular you don't care about the max lvl grid because you are only checking a subtyping relation...